### PR TITLE
Build option to disable fast math

### PR DESCRIPTION
**Pull request by jbobba**

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -166,6 +166,7 @@ option(NGRAPH_LIB_VERSIONING_ENABLE "Enable shared library versioning" FALSE)
 option(NGRAPH_PYTHON_BUILD_ENABLE "Enable build nGraph python package wheel" FALSE)
 option(NGRAPH_PLAIDML_ENABLE "Enable the PlaidML backend" ${PLAIDML_FOUND})
 option(NGRAPH_DISTRIBUTED_ENABLE "Enable distributed training using MLSL/OpenMPI" OFF)
+option(NGRAPH_DISABLE_FAST_MATH "Disable fast math" OFF)
```
Comment (rkimballn1) [inline CMakeLists.txt]: How about `NGRAPH_FAST_MATH_ENABLE` with the default of ON. That way it follows the naming convention of the other flags.

Comment (baojun-nervana) [reply CMakeLists.txt]: would this say we are currently using eigen fast math already?

File: src/ngraph/runtime/cpu/CMakeLists.txt [hunk 2]
```diff
@@ -165,6 +165,10 @@ if (NGRAPH_CPU_ENABLE)
         add_dependencies(cpu_backend ext_halide)
     endif()
 
+    if (NGRAPH_DISABLE_FAST_MATH)
+        target_compile_definitions(cpu_backend PRIVATE EIGEN_FAST_MATH=0)
```
Comment (baojun-nervana) [inline src/ngraph/runtime/cpu/CMakeLists.txt]: I tried to set EIGEN_FAST_MATH=1. It doesn't change anything on the result somehow.

File: CMakeLists.txt (hunk 1 above)
Comment (jbobba) [reply CMakeLists.txt]: yes, we are enabling by default.

**Pull request merged**